polaris-core/src/main/java/org/apache/polaris/core/entity/CatalogEntity.java

```diff
@@ -249,12 +249,19 @@ public Builder setStorageConfigurationInfo(
             break;
           case AZURE:
             AzureStorageConfigInfo azureConfigModel = (AzureStorageConfigInfo) storageConfigModel;
-            config =
+            AzureStorageConfigurationInfo azureconfigInfo =
                 new AzureStorageConfigurationInfo(
                     new ArrayList<>(allowedLocations), azureConfigModel.getTenantId());
+            azureconfigInfo.setMultiTenantAppName(azureConfigModel.getMultiTenantAppName());
```

Comment:
This makes sense, good fine. For AWS, we do it like this with casting:

```
                new AwsStorageConfigurationInfo(
                    PolarisStorageConfigurationInfo.StorageType.S3,
                    new ArrayList<>(allowedLocations),
                    awsConfigModel.getRoleArn(),
                    awsConfigModel.getExternalId());
            ((AwsStorageConfigurationInfo) config).validateArn(awsConfigModel.getRoleArn());
```

Can you maybe also refactor the AWS bit to match this? Or use casting here, either way seems okay to me.

Reply:
For the AWS case, we cast to invoke a single specific method. Here we need to invoke two methods. I think using the inline cast to invoke a single method makes sense and doesn't hurt readability, whereas inline casting twice is unnecessarily cumbersome, hence the intermediate variable

Reply:
> For the AWS case, we cast to invoke a single specific method. Here we need to invoke two methods.

For GCP, it seems there is also a single method invocation